geti_sdk/prediction_visualization/visualizer.py

```diff
@@ -0,0 +1,115 @@
+from geti_sdk.prediction_visualization.shape_drawer import ShapeDrawer
+
+
+class Visualizer:
+    """
+    Visualize the predicted output by drawing the annotations on the input image.
+
+    :example:
+        >>> predictions = inference_model.predict(frame)
+        >>> annotation = prediction_converter.convert_to_annotation(predictions)
+        >>> output = visualizer.draw(frame, annotation.shape, annotation.get_labels())
+        >>> visualizer.show(output)
+
+    """
+
+    def __init__(
+        self,
+        window_name: Optional[str] = None,
+        show_labels: bool = True,
+        show_confidence: bool = True,
+        show_count: bool = False,
+        is_one_label: bool = False,
+        no_show: bool = False,
+        delay: Optional[int] = None,
+        output: Optional[str] = None,
+    ) -> None:
+        """
+        Initialize the Visualizer.
+
+        :param window_name: Name of the window to be shown
+        :param show_labels: Show labels on the output image
+        :param show_confidence: Show confidence on the output image
+        :param show_count: Show count of the shapes on the output image
+        :param is_one_label: Show only one label on the output image
+        :param no_show: Do not show the output image
+        :param delay: Delay time for the output image
+        :param output: Path to save the output image
+        """
+        self.window_name = "Window" if window_name is None else window_name
+        self.shape_drawer = ShapeDrawer(
+            show_count, is_one_label, show_labels, show_confidence
+        )
+
+        self.delay = delay
+        self.no_show = no_show
+        if delay is None:
+            self.delay = 1
+        self.output = output
+
+    def draw(
+        self,
+        image: np.ndarray,
+        annotation: AnnotationScene,
+        fill_shapes: bool = True,
+        confidence_threshold: Optional[float] = None,
+        meta: Optional[dict] = None,
+    ) -> np.ndarray:
+        """
+        Draw annotations on the image.
+
+        :param image: Input image in RGB format
+        :param annotation: Annotations to be drawn on the input image
+        :param meta: Optional meta information
+        :param fill_shapes: Fill shapes with color
+        :param confidence_threshold: Confidence threshold to filter annotations.
+            Must be in range [0, 1].
+        :return: Output image with annotations in RGB format
+        """
+        image = cv2.cvtColor(image, cv2.COLOR_RGB2BGR)
+
+        if confidence_threshold is not None:
+            annotation = annotation.filter_by_confidence(confidence_threshold)
+        result = self.shape_drawer.draw(
+            image, annotation, labels=[], fill_shapes=fill_shapes
+        )
+        return cv2.cvtColor(result, cv2.COLOR_BGR2RGB)
+
+    def show(self, image: np.ndarray) -> None:
+        """
+        Show result image.
+
+        :param image: Image to be shown.
+        """
+        if not self.no_show:
```

Comment:
This seems weird, I'd say we should at least log that `.show` has been called while the interactive output was suppressed. Otherwise it might leave a user wondering why they are not seeing any output.